(In Code Review) Write unit tests for User

jerrellmardis opened the pull request:
https://jira.desk.technology/browse/APIJ-70

Commit:
Added tests for User class.

mattkranzler5 commented on api-client/src/main/java/com/desk/java/apiclient/model/Links.java:
```diff
@@ -74,4 +74,8 @@ public Link getPrevious() {
         return next == null ? new Link() : next;
     }
+
+    void setSelf(Link self) {
```
I was trying to avoid adding a setter for these base links. If this is only for the purpose of the test below I think we should just mock the `userLinks.getSelfLink()` for the test.

mattkranzler5 commented on api-client/src/main/java/com/desk/java/apiclient/model/User.java:
```diff
@@ -52,6 +52,7 @@ public void setId(int i) {
         this.id = i;
     }
 
+    @NotNull
```
For simple getters like this where there are checks like this I've been changing them to just pass the null property through. I think doing that and leaving it up to the client to change to an empty string for consistency and removing logic from the model objects is a good way to go. What do you think?

mattkranzler5 commented on api-client/src/main/java/com/desk/java/apiclient/model/User.java:
```diff
@@ -84,6 +88,7 @@ public void setLevel(String l) {
         this.level = l;
     }
 
+    @NotNull
```
Same as above, I'd prefer to just return null.

mattkranzler5 commented:
@jerrellmardis first pass done.

Commit:
Updated UserTest and User object.

• Now returning null objects for getName and getAvatar
• Removed Links#setSelf(Link) method in favor of mocking the call in UserTest.

mattkranzler5 commented:
👍
